### ISSUE TITLE
Change the name

### ISSUE THREAD

**Issue by n8behavior** [stdlib/lib#150]
I get it, you think it's cute but <stdlib> is already a real thing with many decades of precedence.  It has things like `size_t` and is ACTUALLY and ALREADY the basis for computing today...even "microservices"...yet another buzz word for just-doing-good-software-engineering-for-the-problem-under-discussion.  This project's name reminds be of the choosing of "Javascript" which had (and thankfully so) nothing to do with Java, just tried to leverage the name "Java:tm:".

TL;DR things are confusing enough without folks overloading and stepping on names.  Feel like you could be a little creative and get your own name.

[starts a pool on how fast this issue gets closed]

**Comment by peconi** [comment]
+1

**Comment by keithwhor** [comment]
You're welcome to refer to us as "stdlib.com"! Let's keep it open for a while, I'd put it at 1 in every 100 developers (-ish) we encounter questions the name, we occasionally get a few people a little more passionate (like yourself)! I made a Hacker News post about this recently, which I'm happy to copy and paste here:

> We chose the name as a reference or throwback, of course, to <stdlib.h>. We’re <stdlib.com>, and we allow developers to utilize serverless technology to make shipping custom logic (functions) simple and create a Standard Library of APIs within their organization or contribute to a public library.
From the inception of the product we knew we were building a library - well, a publishing and hosting platform combined - but really, a library of APIs. The very first name we came up with was “stdlib” and, through a stroke of fate, “stdlib.com” was available.
>
> We don’t take the name lightly. We understand we stand on the shoulders of giants, and it is our responsibility to create the best “stdlib.com” we can. It motivates and inspires our team members. That is why we chose the name.
>
> It’s unfortunate you have a problem with this, but we’re very thankful to our developer community, team, partners and our investors for their support. :)

So, the tl;dr is of course: I understand your concern, but the answer is, "we're not changing the name." I'll close the issue if discourse crosses the line of civility (keep it clean, folks!), but hey. You're welcome to your opinion and it can serve as a helpful reference to others. I understand this won't be a favorite answer to the people who agree with you, and I think it's somewhat likely that by keeping this issue *open*, for now, I may be inviting some unwanted attention but...

Maybe we can keep the honeypot sweet, and if there's enough banter figure out ways to encourage some healthy (and honest) discussion. 😊 I would definitely encourage the community to reference us as "stdlib.com" where they believe naming conflicts may cause confusion. (StackOverflow for example -- but we don't have a tag there yet!)

**Comment by keithwhor** [comment]
Also, quick addition:

I think what may be helpful to realize is a significant portion of our developer community are web-native and / or come from less technical backgrounds. We often get questions about the name from these developers and customers who simply *don't understand what stdlib means*. This actually creates a really neat opportunity to [teach people something they don't know](https://xkcd.com/1053/), where we can explain that "stdlib.com" is a reference to the [C Standard Library](https://en.wikipedia.org/wiki/C_standard_library).

A large chunk of what we'd like to accomplish is lowering the barrier to entry for software development *in general* for future knowledge workers. In part, the goal is to help turn remote APIs into [first-class citizens of a developers' environment](https://en.wikipedia.org/wiki/First-class_citizen) (or, head in that direction). It's ambitious, certainly, but we're of the position that we're coming from a deeply respectful place and have the opportunity to introduce people to foundational concepts of software development — well, building APIs — using the web (and "serverless" architectures) as a springboard, while still providing the necessary tooling for veterans of the industry to move quickly from within their local environments.

If anything, consider that we think about the name like this: our goal is to educate, not confuse. Your concerns are healthy, and coming from the right place, but I would argue that based on our experience with both our customers and general community to date that there's room in the ecosystem to share headspace with already existing concepts. We have not, in practice, and through the course of interacting directly with customers and developers, witnessed any actual confusion to date beyond the occasional, "is this like the C standard library?" from a few savvier, but early, software developers at conferences / Hackathons who are quickly receptive to a short explanation.

It's up to us to ensure it stays that way. Which, hey, maybe this issue will help.

**Comment by jamsinclair** [comment]
Are the library/service maintainers aware there is a JavaScript library that predates your `stdlib` over at https://github.com/stdlib-js? (Which is an actual Standard Library written in JavaScript)

I was doing a google search trying to find `stdlib-js` and ended up here all confused 😕. Another point of confusion for your name. 

---

>  I'd put it at 1 in every 100 developers (-ish) we encounter questions the name, we occasionally get a few people a little more passionate (like yourself)! I made a Hacker News post about this recently

> ...the occasional, "is this like the C standard library?" from a few savvier, but early, software developers at conferences / Hackathons who are quickly receptive to a short explanation.

So you're spending extra time to explain, write posts and justify your name choice to new people.

If you didn't have a conflicting name that required explanation, would more time be spent making your platform and service more awesome? 😄

**Comment by keithwhor** [comment]
Hey @jamsinclair -- you'll have to bring up your issues with the `stdlib-js` maintainers if you think there's a conflict. If your argument is simply "we're not a standard library for JavaScript" -- you're right, and we have no intention of being such. We provide an API development platform and library, hence, "Standard Library." The name is both descriptive and useful for us and both new developers and customers, as outlined.

To your second point: of course we're spending time engaging with our community about our product! I think we'd be pretty poor heralds of our vision if we didn't take time to chat with and engage with people. We're fairly passionate about our ambitions, software and community - so it's no sweat off our back to answer questions. 😊

You should definitely try building some APIs with us, while you're here! We'll be releasing some pretty great tools shortly that should help even more, so stay tuned.

**Comment by keithwhor** [comment]
Closing issue for now; useful discussion hasn't been had in months. If it helps anybody, we're going by "Standard Library" now in most of our website documentation. 🙂

**Comment by RoestVrijStaal** [comment]
And from now on each programmer internship student or just hired person wastes one or more days of his first week because his coach, boss or colleague responded with "Look at the standard library" (Often omitting the part "of programming language XYZ") at a trivial question of help 🤦🏻‍🤦🏻‍♂️🤦🏻‍♂️

Congratulations, Well done 👍🏻👍🏻👍🏻

How about to call it StaLi?
